src/core/search/searchdatatransformer.js

```diff
@@ -30,42 +30,46 @@ export default class SearchDataTransformer {
     };
   }
 
-  static transformVertical (data, formatters, verticalKey) {
-    const response = SearchDataTransformer._parseVerticalResponse(data.response);
+  static transformVertical (coreResponse, verticalKey) {
+    const hasResults = coreResponse.verticalResults &&
+      coreResponse.verticalResults.results &&
+      coreResponse.verticalResults.resultsCount > 0;
+
+    let resultsContext = ResultsContext.NORMAL;
+    let response = coreResponse;
+    if (!hasResults) {
+      resultsContext = ResultsContext.NO_RESULTS;
+      response = SearchDataTransformer._reshapeForNoResults(coreResponse);
+    }
+
     return {
       [StorageKeys.QUERY_ID]: response.queryId,
       [StorageKeys.NAVIGATION]: new Navigation(), // Vertical doesn't respond with ordering, so use empty nav.
-      [StorageKeys.VERTICAL_RESULTS]: VerticalResults.from(response, formatters, verticalKey),
-      [StorageKeys.DYNAMIC_FILTERS]: DynamicFilters.from(response),
-      [StorageKeys.INTENTS]: SearchIntents.from(response.searchIntents),
+      [StorageKeys.VERTICAL_RESULTS]: VerticalResults.fromCore(
+        response.verticalResults, {}, resultsContext, verticalKey),
+      [StorageKeys.DYNAMIC_FILTERS]: DynamicFilters.from(response.facets, resultsContext),
+      [StorageKeys.INTENTS]: SearchIntents.fromCore(response.searchIntents),
       [StorageKeys.SPELL_CHECK]: SpellCheck.from(response.spellCheck),
-      [StorageKeys.ALTERNATIVE_VERTICALS]: AlternativeVerticals.from(response, formatters),
-      [StorageKeys.LOCATION_BIAS]: LocationBias.from(response.locationBias)
+      [StorageKeys.ALTERNATIVE_VERTICALS]: AlternativeVerticals.from(response.alternativeVerticals),
+      [StorageKeys.LOCATION_BIAS]: LocationBias.fromCore(response.locationBias)
```

Comment:
I think we should be consistent with the naming of the `from/fromCore` methods. Either they are all called `fromCore` or all called `from` (and it's implicitly understood that we're translating from Core library models).

Reply:
Yeah that sounds good to me! At this point, it looks like there is less code change if I rename all the "from"'s to "fromCore". Updating!